wallet-new/server/Cardano/Wallet/API/V1/LegacyHandlers/Transactions.hs

```diff
@@ -102,3 +102,44 @@ estimateFees Payment{..} = do
     fee <- V0.rewrapTxError "Cannot compute transaction fee" $
         eitherToThrow =<< V0.runTxCreator policy (V0.computeTxFee pendingAddrs utxo outputs)
     single <$> migrate fee
+
+
+newUnsignedTransaction
+    :: forall ctx m . (V0.MonadWalletTxFull ctx m)
+    => Payment
+    -> m (WalletResponse Transaction)
+newUnsignedTransaction pmt@Payment {..} = do
+    -- We're creating new transaction as usually, but we mustn't sign/publish it.
+    -- This transaction will be signed on the client-side (mobile client or
+    -- hardware wallet), and after that transaction (with its signature) will be
+    -- sent to backend.
+    let (V1 spendingPw) = fromMaybe (V1 mempty) pmtSpendingPassword
```

Comment:
This line is a little odd -- I'd write `let spendingPw = maybe mempty unV1 pmtSpendingPassword` to avoid the need for all the V1 stuff